sympy/core/tests/test_args.py

```diff
@@ -60,8 +63,19 @@ def test_all_classes_are_tested():
             if not names:
                 continue
 
+            # aesaracode imports aesara which imports distutils which warns:
+            #
+            # DeprecationWarning: The distutils package is deprecated and
+            # slated for removal in Python 3.12. Use setuptools or check PEP
+            # 632 for potential alternatives
+            if submodule == 'sympy.printing.aesaracode':
+                warnings = DeprecationWarning
+            else:
+                warnings = ()
```

Comment:
Lots of places where we need to suppress warnings from Aesara. This can be reverted if Aesara stops using distutils.

Reply:
I don't think we have an issue for this change, but I can quickly make a PR regardless.

Reply:
There is https://github.com/aesara-devs/aesara/issues/758

Reply:
It's both kinds of distutils :)

The numpy.distutils deprecation is just a bit louder:
```console
$ python -Werror -c 'import aesara'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/aesara/__init__.py", line 126, in <module>
    from aesara import scalar, tensor
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/aesara/scalar/__init__.py", line 1, in <module>
    from .basic import *
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/aesara/scalar/basic.py", line 25, in <module>
    from aesara import printing
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/aesara/printing.py", line 20, in <module>
    from aesara.compile import Function, SharedVariable
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/aesara/compile/__init__.py", line 19, in <module>
    from aesara.compile.mode import (
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/aesara/compile/mode.py", line 28, in <module>
    from aesara.link.c.basic import CLinker, OpWiseCLinker
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/aesara/link/c/basic.py", line 26, in <module>
    from aesara.link.c.cmodule import (
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/aesara/link/c/cmodule.py", line 25, in <module>
    import numpy.distutils
  File "/home/oscar/current/sympy/sympy.git/38venv/lib/python3.8/site-packages/numpy/distutils/__init__.py", line 31, in <module>
    warnings.warn("\n\n"
DeprecationWarning: 

  `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
  of the deprecation of `distutils` itself. It will be removed for
  Python >= 3.12. For older Python versions it will remain present.
  It is recommended to use `setuptools < 60.0` for those Python versions.
  For more details, see:
    https://numpy.org/devdocs/reference/distutils_status_migration.html 

```

Reply:
All right, I added both to that PR.